component/src/test/java/com/yahoo/component/VersionIncompatibilityTest.java

```diff
@@ -0,0 +1,115 @@
+        VersionCompatibility compatibility = VersionCompatibility.fromVersionList(versions);
+        assertTrue(compatibility.accept(new Version(0, 0, 0), new Version(0, 0, 0)));
+        assertTrue(compatibility.accept(new Version(0, 0, 0), new Version(0, 0, 1)));
+        assertTrue(compatibility.accept(new Version(0, 0, 0), new Version(0, 1, 0)));
+        assertTrue(compatibility.accept(new Version(0, 0, 0), new Version(1, 0, 0)));
+    }
+
+    @Test
+    public void testValidIncompatibilities() {
+        List<String> versions = List.of("1.2.*", "2", "3.*", "4.0.0", "4.0.1", "4.0.3", "4.1.0", "5");
+        VersionCompatibility compatibility = VersionCompatibility.fromVersionList(versions);
+        assertTrue (compatibility.accept(new Version(0, 0, 0), new Version(0, 0, 0)));
+        assertTrue (compatibility.accept(new Version(0, 0, 0), new Version(1, 1, 1)));
+        assertFalse(compatibility.accept(new Version(0, 0, 0), new Version(1, 2, 3)));
+        assertFalse(compatibility.accept(new Version(1, 1, 0), new Version(1, 2, 0)));
+        assertFalse(compatibility.accept(new Version(1, 2, 1), new Version(1, 2, 0)));
+        assertFalse(compatibility.accept(new Version(1, 1, 0), new Version(1, 3, 0)));
+        assertTrue (compatibility.accept(new Version(1, 2, 3), new Version(1, 2, 3)));
+        assertTrue (compatibility.accept(new Version(1, 3, 0), new Version(1, 9, 9)));
+        assertFalse(compatibility.accept(new Version(1, 3, 0), new Version(2, 0, 0)));
+        assertTrue (compatibility.accept(new Version(2, 0, 0), new Version(2, 2, 2)));
+        assertFalse(compatibility.accept(new Version(2, 0, 0), new Version(3, 0, 0)));
+        assertTrue (compatibility.accept(new Version(3, 0, 0), new Version(3, 0, 0)));
+        assertFalse(compatibility.accept(new Version(3, 0, 0), new Version(3, 1, 0)));
+        assertTrue (compatibility.accept(new Version(3, 0, 0), new Version(3, 0, 1)));
+        assertFalse(compatibility.accept(new Version(3, 0, 0), new Version(4, 0, 0)));
+        assertFalse(compatibility.accept(new Version(4, 0, 0), new Version(4, 0, 1)));
+        assertTrue (compatibility.accept(new Version(4, 0, 1), new Version(4, 0, 2)));
+        assertFalse(compatibility.accept(new Version(4, 0, 2), new Version(4, 0, 3)));
+        assertFalse(compatibility.accept(new Version(4, 0, 3), new Version(4, 1, 0)));
+        assertFalse(compatibility.accept(new Version(4, 1, 0), new Version(5, 0, 0)));
+        assertTrue (compatibility.accept(new Version(5, 0, 0), new Version(6, 0, 0)));
+        assertFalse(compatibility.accept(new Version(0, 0, 0), new Version(2, 0, 0)));
+        assertFalse(compatibility.accept(new Version(0, 0, 0), new Version(6, 0, 0)));
+    }
+
+    @Test
+    public void testIllegalIncompatibilities() {
+        try {
+            VersionCompatibility.fromVersionList(List.of("1", "*"));
+            fail();
+        }
+        catch (IllegalArgumentException expected) { }
```

Comment:
Consider extracting a method for this pattern.

Reply:
Hmm yes.